### ISSUE TITLE
[Accessibility]: Search result type no longer read by screen reader

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#43142]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34190

### Problem description

Since we removed the search result type icon, the result type is no longer being read by VoiceOver (it was the icon's label). This makes the search a lot less scannable, especially in [landmarks mode](https://twitter.com/sourcegraph/status/1541892998270795776). 

### Expected behavior

Search result type is read by screen reader as part of the result's label (and it shows up in the landmarks list).

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
